spec/02-integration/000-new-dao/02-db_core_entities_spec.lua

```diff
@@ -410,10 +430,15 @@ for _, strategy in helpers.each_strategy() do
             code        = Errors.codes.SCHEMA_VIOLATION,
             name = "schema violation",
             strategy    = strategy,
-            message     = "schema violation (paths: field required for entity check)",
-            fields      = {
-              paths = "field required for entity check",
-            }
+            message  = unindent([[
+              schema violation
+              (at least one of 'methods', 'hosts' or 'paths' must be non-empty)
+            ]], true, true),
+            fields   = {
+              ["@entity"] = {
+                "at least one of 'methods', 'hosts' or 'paths' must be non-empty",
+              }
+            },
```

Comment:
It seems like this test becomes the same one as the above "fails trying to unset an interdependent field by itself":

1. we create a valid entity
2. we send an update with all included "semi-required" fields set to null

It seems like it becomes "fails trying to unset many interdependent field by themselves", which feels redundant with the previous test? It only differs in what the original entity has prior to the update, which should not change the result of the test because it focuses on the update operation.

The above test also deserves a comment that I will add here for brevity: it creates an entity with both `hosts` and `methods`, and sends an update nullifying `methods`. But the received error would be hard to swallow as a user: `at least one of 'methods', 'hosts' or 'paths' must be non-empty`. Yes, that is still the case. But the user does not know we do not do a read-before-write when updating this entity. It might be worth returning:

```lua
{
  ["@entity"] = "at least one of '', '', '' must be non-empty when updating",
  methods = "field required for entity check when updating",
  hosts = "field required for entity check when updating",
  paths = "field required for entity check when updating",
}
```

In other words, I think the behavior would be:
- In insertion
  - always require at least one of the three
  - error out with the "at least one of..."
- In update
  - if one of the required values is `nil` or `null`, require a non-null value
  - if all values are `nil` or `null`, error out with the more descriptive "field required for entity check when updating" for the user to be less confused
- Remove the existing "field required for entity check" error seems it seems it comes unused? Or update it to the above, more descriptive, update-aware message

This ended up being a bit longer than I thought, but I hope it makes sense.

Reply:
I definitely appreciate the thought given to the matter!

I modified the behavior to match what you described (removing the existing "field required for entity check" error in those cases, since the overall entity message now does a better job at describing what's going on). I also renamed the test, your description is better.

I got a bit confused about the redundancy of the tests -- I added it because the existing ones didn't cover the behavior I wanted to change. Perhaps it now covers both the behavior I wanted to test and that of another test? In other words, which particular test are you asking to be removed?